### PR TITLE
Updates stat_tracker and initialize testing for all classes, add module, iteration 3 methods

### PR DESCRIPTION
**Pull request by rer7891**
Winningest team and worst fans are not passing my rake but that is it. I believe those are both wren's so that should be fixed with her update. If you want me to do that before approving this let me know.

### CODE REVIEW

File: lib/team_collection.rb [hunk 1]
```diff
@@ -17,4 +17,8 @@ def all_teams
     end
     team_objects
   end
+
+  def highest_scoring_visitor
+  x = @game_instances.highest_visitor_score
```
Comment (PaulDebevec) [inline lib/team_collection.rb]: Hey, can we change 'x' into a descriptive variable?

**Review by PaulDebevec — CHANGES_REQUESTED** [review]
All I saw that needs to be changed is that variable name, and after that change, I think it's good to merge!

File: lib/team_collection.rb (hunk 1 above)
Comment (rer7891) [reply lib/team_collection.rb]: I actually don't see this in my code?? At the top of the file it says outdated. What does that mean? I'm confused where that is coming from. I double checked and it's not in my branch file.

Comment (PaulDebevec) [reply lib/team_collection.rb]: Oh ^^ I'm seeing it's actually in our Master branch... I changed it in mine and I'll make the change with my PR

**Review by PaulDebevec — APPROVED** [review]

**Comment by wrenisit** [comment]
This looks great! I love the incorporation of the module for the helper methods.

**Pull request merged**